src/components/ThemeContext.ts

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+const noOp = () => {};
+
+export type UiMode = 'light' | 'dark';
+
+export interface ThemeContext {
+    uiMode: UiMode;
+    setUiMode: (newMode: UiMode) => void; 
+}
+
+const themeContext = React.createContext<ThemeContext>({
+    uiMode: 'light',
+    setUiMode: noOp,
```

Comment:
Instead of creating a noop handler, we should create a function that logs warning that the default function is being used.